plugins/inputs/openweathermap/openweathermap.go

```diff
@@ -0,0 +1,305 @@
+					strs = append(strs, n.CityId[j])
+					j++
+				}
+				cities := strings.Join(strs, ",")
+
+				u, err = url.Parse(fmt.Sprintf("/data/2.5/group?id=%s&APPID=%s&units=%s", cities, n.AppId, units))
+				if err != nil {
+					acc.AddError(fmt.Errorf("Unable to parse address '%s': %s", u, err))
+					continue
+				}
+
+				addr = base.ResolveReference(u)
+				wg.Add(1)
+				go func(addr *url.URL) {
+					defer wg.Done()
+					acc.AddError(n.gatherUrl(addr, acc, false))
+				}(addr)
+			}
+
+		}
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func (n *OpenWeatherMap) createHttpClient() (*http.Client, error) {
+
+	if n.ResponseTimeout.Duration < time.Second {
+		n.ResponseTimeout.Duration = defaultResponseTimeout
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{},
+		Timeout:   n.ResponseTimeout.Duration,
+	}
+
+	return client, nil
+}
+
+func (n *OpenWeatherMap) gatherUrl(addr *url.URL, acc telegraf.Accumulator, forecast bool) error {
+	resp, err := n.client.Get(addr.String())
+
+	if err != nil {
+		return fmt.Errorf("error making HTTP request to %s: %s", addr.String(), err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("%s returned HTTP status %s", addr.String(), resp.Status)
+	}
+	contentType := strings.Split(resp.Header.Get("Content-Type"), ";")[0]
+	switch contentType {
+	case "application/json":
+		err = gatherWeatherUrl(bufio.NewReader(resp.Body), forecast, acc)
+		return err
+	default:
+		return fmt.Errorf("%s returned unexpected content type %s", addr.String(), contentType)
+	}
+}
+
+type WeatherEntry struct {
+	Dt     int64  `json:"dt"`
+	Dttxt  string `json:"dt_txt"` // empty for weather/
+	Clouds struct {
+		All int64 `json:"all"`
+	} `json:"clouds"`
+	Main struct {
+		GrndLevel float64 `json:"grnd_level"` // empty for weather/
+		Humidity  int64   `json:"humidity"`
+		SeaLevel  float64 `json:"sea_level"` // empty for weather/
+		Pressure  float64 `json:"pressure"`
+		Temp      float64 `json:"temp"`
+		TempMax   float64 `json:"temp_max"`
+		TempMin   float64 `json:"temp_min"`
+	} `json:"main"`
+	Rain struct {
+		Rain3 float64 `json:"3h"`
+	} `json:"rain"`
+	Sys struct {
+		Pod     string  `json:"pod"`
+		Country string  `json:"country"`
+		Message float64 `json:"message"`
+		Id      int64   `json:"id"`
+		Type    int64   `json:"type"`
+		Sunrise int64   `json:"sunrise"`
+		Sunset  int64   `json:"sunset"`
+	} `json:"sys"`
+	Wind struct {
+		Deg   float64 `json:"deg"`
+		Speed float64 `json:"speed"`
+	} `json:"wind"`
+	Weather []struct {
+		Description string `json:"description"`
+		Icon        string `json:"icon"`
+		Id          int64  `json:"id"`
+		Main        string `json:"main"`
+	} `json:"weather"`
+
+	// Additional entries for weather/
+	Id    int64  `json:"id"`
+	Name  string `json:"name"`
+	Coord struct {
+		Lat float64 `json:"lat"`
+		Lon float64 `json:"lon"`
+	} `json:"coord"`
+	Visibility int64 `json:"visibility"`
+}
+
+type Status struct {
```

Comment:
Remove `Coord` and `Country`as they are unused. Also `Name` if it's not going to be added as a tag (could always be removed with [`tagexclude`](https://github.com/influxdata/telegraf/blob/master/docs/CONFIGURATION.md#modifiers)).